File: python/22.py

st = ''
with open("p022_names.txt", "r") as f:
    st = f.read().split(",")
    st = [x[1:len(x)-1] for x in st]
    st.sort()

alphabet = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
name_weight = {}
for i, x in enumerate(alphabet):
    name_weight[x] = i+1
count = 0
for i in range(len(st)):
    name = st[i]
    k = 0
    for j in name:
        k += name_weight[j]
    count += k*(i+1)

print(count)

File: cpp/22.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <sstream>
#include <algorithm>
#include <map>

using namespace std;

vector<string> split(const string s, char delimiter=',')
{
   vector<string> tokens;
   string token;
   istringstream tokenStream(s);
   while (getline(tokenStream, token, delimiter))
   {
      tokens.push_back(token.substr(1, token.length() - 2));
   }
   return tokens;
}

int main() {
    long int count = 0;

    fstream file("22.txt");
    string names;
    getline(file, names);

    vector<string> vect_names = {};
    vect_names = split(names);
    sort(vect_names.begin(), vect_names.end());

    string alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    map<char, int> name_weight {};
    int i = 1;
    for (char c: alphabet){
        name_weight[c] = i;
        i++;
    }

    i = 1;
    for (string name: vect_names)
    {
        int weight = 0;
        for (char c: name)
            weight += name_weight[c];
        
        count += weight * i;
        i++;
    }
    
    cout << count << endl;
    return 0;
}